### PR TITLE
Favorites functionality is affecting the next/previous overlay functionality

### PR DESCRIPTION
**Pull request by yairtawil**

### CODE REVIEW

File: src/app/@ansyn/core/utils/sorting.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 // latest date first
-export const sortByDateDesc = (o1, o2) => o1.date > o2.date ? -1 : 1;
+export const sortByDateDesc = (o1, o2) => new Date(o1.date) > new Date(o2.date) ? 1 : -1;
```
Comment (liorze) [inline src/app/@ansyn/core/utils/sorting.ts]: `new Date(o1.date) - new Date(o2.date)`

File: src/app/@ansyn/core/utils/sorting.ts [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
+export const sortByDate = (o1, o2) => new Date(o2.date) > new Date(o1.date) ? 1 : -1;
```
Comment (liorze) [inline src/app/@ansyn/core/utils/sorting.ts]: `-sortByDateDesc(o1, o2)`

File: src/app/@ansyn/overlays/services/overlays.service.ts [hunk 3]
```diff
@@ -7,46 +7,23 @@ import { OverlaysCriteria, OverlaysFetchData } from '@ansyn/core/models/overlay.
...
 export const OverlaysConfig: InjectionToken<IOverlaysConfig> = new InjectionToken('overlays-config');
 
 @Injectable()
 export class OverlaysService {
 
-	static filter(overlays: Map<string, Overlay>, filters: { key: any, filterFunc: (ovrelay: any, key: string) => boolean }[]): string[] {
-		if (!overlays) {
-			return [];
+	static buildFilteredOverlays(overlays: Overlay[], parsedFilters: FilterModel[], favorites: FavoritesModel): string[] {
+		let parseedOverlays: Overlay[] = favorites.overlays;
```
Comment (liorze) [inline src/app/@ansyn/overlays/services/overlays.service.ts]: s/parseedOverlays/parsedOverlays

File: src/app/@ansyn/overlays/components/timeline/timeline.component.ts [hunk 4]
```diff
@@ -140,22 +139,19 @@ export class TimelineComponent implements OnInit, OnDestroy {
...
-			name: entities.name || '',
-			data: entities.data
-		})))
-		.filter((drops) => Boolean(drops && drops[0] && drops[0].data))
+	drops: OverlayDrop[] = [];
+	dropsChange$: Observable<OverlayDrop[]> = this.store$.select(selectDrops)
+		.filter(Boolean)
 		.do(drops => {
-			drops[0].data.forEach(drop => {
+			this.dropsIdMap = new Map();
```
Comment (liorze) [inline src/app/@ansyn/overlays/components/timeline/timeline.component.ts]: You can use groupBy here.

File: src/app/@ansyn/menu-items/filters/services/filters.service.ts [hunk 5]
```diff
@@ -31,6 +32,13 @@ export class FiltersService {
 		return { showOnlyFavorites, filters };
 	}
 
+	static pluckFilterModels({ filters }: IFiltersState): FilterModel[] {
+		return <any> Array.from(filters).map(([key, value]): FilterModel => ({
```
Comment (liorze) [inline src/app/@ansyn/menu-items/filters/services/filters.service.ts]: Why do you need the cast to any here?

File: src/app/@ansyn/menu-items/filters/reducer/filters.reducer.ts [hunk 6]
```diff
@@ -15,6 +19,10 @@ export const initialFiltersState: IFiltersState = {
 	filters: new Map<Filter, FilterMetadata>(),
 	oldFilters: null,
 	isLoading: true,
+	facets: {
```
Comment (shaharido1) [inline src/app/@ansyn/menu-items/filters/reducer/filters.reducer.ts]: delete

**Pull request merged**